### PR TITLE
Allow players to un-ignore previously ignored popups for ghostroles, and ignore notifications with no ignore button

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
:cl: Naksu
add: Added a new ghost verb that lets you change your ignore settings, allowing previously ignored popups to be un-ignored and notifications without an ignore button to be ignored
/:cl:

image

### CODE REVIEW

**Comment by Ratvar** [comment]
where's "Do you want to play as Rob Ust? Job: Tator toe" preference, tho

**Comment by BeeSting12** [comment]
I love you

**Comment by Shadowflame909** [comment]
WHERES THE MINEBOT OPTION. Accidentally clicking yes to it is the worst thing ever reeeeeeeeeee

**Comment by vuonojenmustaturska** [comment]
minebot is sentience potion

File: code/modules/mob/dead/observer/notificationprefs.dm [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	set name = "Notification preferences"
+	set desc = "Notification preferences"
+
+	var/datum/notificationpanel/panel  = new(usr)
+
+	panel.ui_interact(usr)
+
+
+
+datum/notificationpanel
```
Comment (ninjanomnom) [inline code/modules/mob/dead/observer/notificationprefs.dm]: /

File: code/modules/mob/dead/observer/notificationprefs.dm [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	var/client/user
+
+datum/notificationpanel/New(user)
```
Comment (ninjanomnom) [inline code/modules/mob/dead/observer/notificationprefs.dm]: /

File: code/modules/mob/dead/observer/notificationprefs.dm [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	if (istype(user, /mob))
+		var/mob/M = user
+		if (!M.client)
+			CRASH("Ghost role notification panel attempted to open to a mob without a client")
+		src.user = M.client
+	else
+		src.user = user
+
+datum/notificationpanel/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.observer_state)
```
Comment (ninjanomnom) [inline code/modules/mob/dead/observer/notificationprefs.dm]: /

File: code/modules/mob/dead/observer/notificationprefs.dm [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "notificationpanel", "Notification Preferences", 700, 700, master_ui, state)
+		ui.open()
+
+datum/notificationpanel/ui_data(mob/user)
```
Comment (ninjanomnom) [inline code/modules/mob/dead/observer/notificationprefs.dm]: /

File: code/modules/mob/dead/observer/notificationprefs.dm [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+	panel.ui_interact(usr)
+
+
+
+datum/notificationpanel
+	var/client/user
+
+datum/notificationpanel/New(user)
+	if (istype(user, /mob))
```
Comment (ninjanomnom) [inline code/modules/mob/dead/observer/notificationprefs.dm]: ismob

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**